Sources/NIO/ByteBuffer-views.swift

```diff
@@ -16,48 +16,161 @@
 ///
 /// A `ByteBufferView` is useful whenever a `Collection where Element == UInt8` representing a portion of a
 /// `ByteBuffer` is needed.
-public struct ByteBufferView: RandomAccessCollection {
+public struct ByteBufferView: RandomAccessCollection, MutableCollection, RangeReplaceableCollection {
```

Comment:
I think it would be better to put the `MutableCollection, RangeReplaceableCollection` conformances into extensions instead of directly onto the type?